### ISSUE TITLE
Bug:  Primary key column not recognised by Starlette-admin 

### ISSUE THREAD

**Issue by agxc** [jowilf/starlette-admin#347]
**Describe the bug**

I need to connect to an existing table in the MS SQL server. The convention for these SQL server tables is to separate names like [Column name 1], [Column name 2] etc. These are existing tables in the server that cannot be easily changed. 

However, the starlette-admin doesn't recognise primary key column with this naming convention and renders empty table.

Version 1 (not working)
```py
class TestTable(Base):
    __table_args__  = {"schema": "TestSchema"}
    __tablename__   = "TestTable"
    primary_key_1    = Column( "Primary key 1",   Integer,  primary_key=True)
    column_name_1    = Column("Column name 1", String(1))
    column_name_2    = Column("Column name 2", String(3))
```
When the column name is change from `Primary key 1` to `primary_key_1`, the starlette can properly display records again. 

Version 2 (working)
```py
class TestTable(Base):
    __table_args__  = {"schema": "TestSchema"}
    __tablename__   = "TestTable"
    primary_key_1    = Column(  Integer,  primary_key=True)
    column_name_1    = Column("Column name 1", String(1))
    column_name_2    = Column("Column name 2", String(3))
```

However, I cannot use the Version 2 since I have to connect to large number of tables already existed in the server with this kind of naming convention. Names separated by space is commonly adopted by SQL server users. e.g. [Name 1], [Name 2], [Name 3]. 

**Impact**  :  **High**
Without this bug fixed, I cannot use starlette-admin to connect to any SQL server tables. 

**To Reproduce**
The bug can be reproduced by specifying a primary key name separated by space. I tested the version 1 is working with `Flask-admin`

**Environment (please complete the following information):**
 - Starlette-Admin version: starlette-admin==0.11.2
 - ORM/ODMs: SQLAlchemy, 
 - Database: MS SQL server

**Comment by hasansezertasan** [comment]
Could you try this: [Extending BaseModelView - Starlette Admin](https://jowilf.github.io/starlette-admin/advanced/base-model-view/#primary-key)

**Comment by agxc** [comment]
Thanks for your suggestion @hasansezertasan. Unfortunately, this doesn't resolve my issue. I need `starlett-admin` be able to read column PK named like `primary key 1` instead of `primary_key_1`.  The issue I have here is that there is an existing large tables with columns named with spaces, for example, `Primary key 1`. 

 Upon a closer investigation, I find the problem is not just impacting SQL server but other databases as well (e.g. SQLite as shown in the example code below)

Below is a self-contain app highlighting the issue. You can switch between SQL server and SQLite (#11 and #13) to see the issue. The moment I switch from line #17 to #16, the starlette-admin no longer displays records. 


```py
from sqlalchemy import create_engine
from sqlalchemy import  Column,  String, Integer
from starlette.applications import Starlette
from starlette.routing import  Route
from sqlalchemy.ext.declarative import declarative_base
from starlette.responses import HTMLResponse
from starlette_admin.contrib.sqla import Admin, ModelView
from sqlalchemy.orm import sessionmaker

Base = declarative_base()
engine = create_engine("mssql+pyodbc:///?odbc_connect=DRIVER={ODBC Driver 17 for SQL Server};SERVER=.\SQLEXPRESS;DATABASE=TEST_DB;Trusted_Connection=yes;"
                       , echo=False)
# engine = create_engine("sqlite:///demo.db?check_same_thread=false", echo=False)
class TestTable(Base):
    # __table_args__  = {"schema": "TestSchema"}
    __tablename__   = "TestTable"
    primary_key_1    = Column( "Primary key 1",   Integer,  primary_key=True)
    # primary_key_1    = Column(    Integer,  primary_key=True)
    column_name_1    = Column("Column name 1", String(1))
    column_name_2    = Column("Columna name 2", String(3))

class TestTableView(ModelView):
    page_size_options = [5, 10, 25, -1]


def init_database() -> None:
    Base.metadata.create_all(engine)

json_data = {
    "TestTable": [
        {
            "primary_key_1": "1",
            "column_name_1": "E",
            "column_name_2": "FOO",

        },
        {
            "primary_key_1": "2",
            "column_name_1": "D",
            "column_name_2": "BAR",

        }
    ],

}
data = [TestTable(**x) for x in json_data["TestTable"]]

SessionLocal = sessionmaker(bind=engine)
with SessionLocal() as session:
    Base.metadata.drop_all(engine)
    Base.metadata.create_all(engine)
    session.add_all(data)
    session.commit()


app = Starlette(
    routes=[
        Route(
            "/",
            lambda r: HTMLResponse('<a href="/admin/">Test admin page</a>'),
        )
    ],
    on_startup=[init_database],
)

admin = Admin(engine, title="Example: SQLAlchemy for MS SQL Server")

admin.add_view(TestTableView(TestTable,  icon="fa fa-tag"))
admin.mount_to(app)
```

**Comment by hasansezertasan** [comment]
`starlette_admin.contrib.sqla.view.py:68` extracts the `pk_attr` but even if I set, it still extracts the value.

Go this way:

```python
from typing import Any, Optional, Type

from sqlalchemy import Column, Integer, String, create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker
from starlette.applications import Starlette
from starlette.responses import HTMLResponse
from starlette.routing import Route
from starlette_admin.contrib.sqla import Admin, ModelView
from starlette_admin.contrib.sqla.converters import BaseSQLAModelConverter

Base = declarative_base()
engine = create_engine("sqlite:///./db.sqlite3", echo=False)


# engine = create_engine("sqlite:///demo.db?check_same_thread=false", echo=False)
class TestTable(Base):
    # __table_args__  = {"schema": "TestSchema"}
    __tablename__ = "TestTable"
    primary_key_1 = Column("Primary key 1", Integer, primary_key=True)
    # primary_key_1    = Column(    Integer,  primary_key=True)
    column_name_1 = Column("Column name 1", String(1))
    column_name_2 = Column("Columna name 2", String(3))


class CustomModelView(ModelView):
    def __init__(
        self,
        model: type,
        pk_attr: str | None = None,
        icon: str | None = None,
        name: str | None = None,
        label: str | None = None,
        identity: str | None = None,
        converter: BaseSQLAModelConverter | None = None,
    ):
        super().__init__(model, icon, name, label, identity, converter)
        self.pk_attr = pk_attr


class TestTableView(CustomModelView):
    exclude_fields_from_create = ["primary_key_1"]
    exclude_fields_from_edit = ["primary_key_1"]


def init_database() -> None:
    Base.metadata.create_all(engine)


json_data = {
    "TestTable": [
        {
            "primary_key_1": "1",
            "column_name_1": "E",
            "column_name_2": "FOO",
        },
        {
            "primary_key_1": "2",
            "column_name_1": "D",
            "column_name_2": "BAR",
        },
    ],
}
data = [TestTable(**x) for x in json_data["TestTable"]]

SessionLocal = sessionmaker(bind=engine)
with SessionLocal() as session:
    Base.metadata.drop_all(engine)
    Base.metadata.create_all(engine)
    session.add_all(data)
    session.commit()


app = Starlette(
    routes=[
        Route(
            "/",
            lambda r: HTMLResponse('<a href="/admin/">Test admin page</a>'),
        )
    ],
    on_startup=[init_database],
)

admin = Admin(engine, title="Example: SQLAlchemy for MS SQL Server")

admin.add_view(TestTableView(TestTable, icon="fa fa-tag", pk_attr="primary_key_1"))
admin.mount_to(app)

```

It works but you need to exclude `primary_key_1` from edit and create because it automatically includes it. Be careful with your data, I'm still learning the codebase and not sure what I'm doing. Test it as much as you can before using it in production. I don't want you to lose data.

You better wait for @jowilf's response.

BTW, overriding `pk_attr` doesn't seem to work, we probable should replace `starlette_admin.contrib.sqla.view.py:68` with `self.pk_attr = self._pk_column.key if self.pk_attr is None else self.pk_attr`

**Comment by agxc** [comment]
Thanks @hasansezertasan . You've been very helpful. I can see your workaround can mitigate the issue with SQLite but the moment I switch back to SQL server, your proposed approached encounters other errors as shown below. In any case, we shouldn't expect users to use these workarounds in the first place. 

I've started trying to refactor a flask-admin based app to starlette-admin, but I won't use anything in prod yet. I agree with you that it is best to wait for @jowilf's response then. 

Here is the stacktrace when I switch your code's database uri from SQLite to SQL server. 

```bash
INFO:     Started server process [17104]
INFO:     Waiting for application startup.
2023-10-22 19:35:33,802 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2023-10-22 19:35:33,802 INFO sqlalchemy.engine.Engine SELECT [INFORMATION_SCHEMA].[TABLES].[TABLE_NAME]
FROM [INFORMATION_SCHEMA].[TABLES]
WHERE ([INFORMATION_SCHEMA].[TABLES].[TABLE_TYPE] = CAST(? AS NVARCHAR(max)) OR [INFORMATION_SCHEMA].[TABLES].[TABLE_TYPE] = CAST(? AS NVARCHAR(max))) AND [INFORMATION_SCHEMA].[TABLES].[TABLE_NAME] = CAST(? AS NVARCHAR(max)) AND [INFORMATION_SCHEMA].[TABLES].[TABLE_SCHEMA] = CAST(? AS NVARCHAR(max))
2023-10-22 19:35:33,803 INFO sqlalchemy.engine.Engine [cached since 0.02747s ago] ('BASE TABLE', 'VIEW', 'TestTable', 'dbo')
2023-10-22 19:35:33,803 INFO sqlalchemy.engine.Engine COMMIT
INFO:     Application startup complete.
INFO:     127.0.0.1:54328 - "GET /test-table/list HTTP/1.1" 200 OK
2023-10-22 19:35:55,681 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2023-10-22 19:35:55,682 INFO sqlalchemy.engine.Engine ROLLBACK
INFO:     127.0.0.1:54328 - "GET /api/test-table?skip=0&limit=10 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_utils.py", line 82, in collapse_excgroups
  |     yield
  |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 190, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\_backends\_asyncio.py", line 664, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 408, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    |     return await self.app(scope, receive, send)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\applications.py", line 116, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 746, in __call__
    |     await route.handle(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 459, in handle
    |     await self.app(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\applications.py", line 116, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 189, in __call__
    |     with collapse_excgroups():
    |   File "C:\Users\gcao\AppData\Local\Programs\Python\Python311\Lib\contextlib.py", line 155, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_utils.py", line 88, in collapse_excgroups
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 191, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\contrib\sqla\middleware.py", line 28, in dispatch
    |     return await call_next(request)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 165, in call_next
    |     raise app_exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 151, in coro
    |     await self.app(scope, receive_or_disconnect, send_no_error)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 746, in __call__
    |     await route.handle(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 288, in handle
    |     await self.app(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 75, in app
    |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    |     raise exc
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 70, in app
    |     response = await func(request)
    |                ^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\base.py", line 282, in _render_api
    |     items = await model.find_all(
    |             ^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\contrib\sqla\view.py", line 228, in find_all
    |     (await anyio.to_thread.run_sync(session.execute, stmt))
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    |     return await get_async_backend().run_sync_in_worker_thread(
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\_backends\_asyncio.py", line 2106, in run_sync_in_worker_thread
    |     return await future
    |            ^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\_backends\_asyncio.py", line 833, in run
    |     result = context.run(func, *args)
    |              ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2306, in execute
    |     return self._execute_internal(
    |            ^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2188, in _execute_internal
    |     result: Result[Any] = compile_state_cls.orm_execute_statement(
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\context.py", line 293, in orm_execute_statement
    |     result = conn.execute(
    |              ^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1416, in execute
    |     return meth(
    |            ^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 516, in _execute_on_connection
    |     return connection._execute_clauseelement(
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1631, in _execute_clauseelement
    |     compiled_sql, extracted_params, cache_hit = elem._compile_w_cache(
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 704, in _compile_w_cache
    |     compiled_sql = self._compiler(
    |                    ^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 316, in _compiler
    |     return dialect.statement_compiler(dialect, self, **kw)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2037, in __init__
    |     super().__init__(*args, **kwargs)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 1426, in __init__
    |     Compiled.__init__(self, dialect, statement, **kwargs)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 867, in __init__
    |     self.string = self.process(self.statement, **compile_kwargs)
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 912, in process
    |     return obj._compiler_dispatch(self, **kwargs)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\visitors.py", line 143, in _compiler_dispatch
    |     return meth(self, **kw)  # type: ignore  # noqa: E501
    |            ^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 4730, in visit_select
    |     text = self._compose_select_body(
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 4916, in _compose_select_body
    |     text += self._row_limit_clause(select, **kwargs)
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2167, in _row_limit_clause
    |     self._check_can_use_fetch_limit(select)
    |   File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2145, in _check_can_use_fetch_limit     
    |     raise exc.CompileError(
    | sqlalchemy.exc.CompileError: MSSQL requires an order_by when using an OFFSET or a non-simple LIMIT clause
    +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\applications.py", line 116, in __call__
    await self.middleware_stack(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 746, in __call__
    await route.handle(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 459, in handle
    await self.app(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\applications.py", line 116, in __call__
    await self.middleware_stack(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 189, in __call__
    with collapse_excgroups():
  File "C:\Users\gcao\AppData\Local\Programs\Python\Python311\Lib\contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_utils.py", line 88, in collapse_excgroups
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 191, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\contrib\sqla\middleware.py", line 28, in dispatch
    return await call_next(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 165, in call_next
    raise app_exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\base.py", line 151, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 746, in __call__
    await route.handle(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 75, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette\routing.py", line 70, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\base.py", line 282, in _render_api
    items = await model.find_all(
            ^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\starlette_admin\contrib\sqla\view.py", line 228, in find_all
    (await anyio.to_thread.run_sync(session.execute, stmt))
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\_backends\_asyncio.py", line 2106, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\anyio\_backends\_asyncio.py", line 833, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2306, in execute
    return self._execute_internal(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2188, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\orm\context.py", line 293, in orm_execute_statement
    result = conn.execute(
             ^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1416, in execute
    return meth(
           ^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 516, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1631, in _execute_clauseelement
    compiled_sql, extracted_params, cache_hit = elem._compile_w_cache(
                                                ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 704, in _compile_w_cache
    compiled_sql = self._compiler(
                   ^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 316, in _compiler
    return dialect.statement_compiler(dialect, self, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2037, in __init__
    super().__init__(*args, **kwargs)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 1426, in __init__
    Compiled.__init__(self, dialect, statement, **kwargs)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 867, in __init__
    self.string = self.process(self.statement, **compile_kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 912, in process
    return obj._compiler_dispatch(self, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\visitors.py", line 143, in _compiler_dispatch
    return meth(self, **kw)  # type: ignore  # noqa: E501
           ^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 4730, in visit_select
    text = self._compose_select_body(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\sql\compiler.py", line 4916, in _compose_select_body
    text += self._row_limit_clause(select, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2167, in _row_limit_clause
    self._check_can_use_fetch_limit(select)
  File "D:\codelib\virtualenv\data-viewer-app-test\venv\Lib\site-packages\sqlalchemy\dialects\mssql\base.py", line 2145, in _check_can_use_fetch_limit
    raise exc.CompileError(
sqlalchemy.exc.CompileError: MSSQL requires an order_by when using an OFFSET or a non-simple LIMIT clause
```

**Comment by jowilf** [comment]
Thank you, @hasansezertasan, for your support. This is solved now in #361

**Comment by agxc** [comment]
Thank you @jowilf ..!
